packages/entity-server/src/app/createApp.ts

```diff
@@ -3,53 +3,30 @@
  * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
  */
 
-import express, { Request, Response, NextFunction } from 'express';
-import cors from 'cors';
-import bodyParser from 'body-parser';
-import errorHandler from 'api-error-handler';
-
-import { Authenticator } from '@tupaia/auth';
-import { EntityServerModelRegistry } from '../types';
-
-import { addRoutesToApp } from './addRoutesToApp';
+import { MicroServiceApiBuilder, handleWith } from '@tupaia/server-boilerplate';
+import { TupaiaDatabase } from '@tupaia/database';
+import {
+  HierarchyRequest,
+  SingleEntityRoute,
+  DescendantsRequest,
+  EntityDescendantsRoute,
+} from '../routes';
+import { attachContext } from '../routes/hierarchy/middleware';
 
 /**
  * Set up express server with middleware,
  */
-export function createApp(models: EntityServerModelRegistry) {
-  const app = express();
-
-  /**
-   * Add middleware
-   */
-  app.use(
-    cors({
-      origin: true,
-      credentials: true, // withCredentials needs to be set for cookies to save @see https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/withCredentials
-    }),
-  );
-  app.use(bodyParser.json({ limit: '50mb' }));
-  app.use(errorHandler());
-
-  /**
-   * Add singletons to be attached to req for every route
-   */
-  const authenticator = new Authenticator(models);
-  app.use((req: Request, res: Response, next: NextFunction) => {
-    req.models = models;
-    req.authenticator = authenticator;
-
-    const context = {}; // context is shared between request and response
-    req.context = context;
-    res.context = context;
-
-    next();
-  });
-
-  /**
-   * Add all routes to the app
-   */
-  addRoutesToApp(app);
-
-  return app;
+export function createApp() {
+  return new MicroServiceApiBuilder(new TupaiaDatabase())
+    .useBasicBearerAuth('entity-server')
+    .use<HierarchyRequest>('/v1/hierarchy/:hierarchyName/:entityCode', attachContext)
```

Comment:
Technically it may be possible to do away with the explicit typecasting here because we ought to be able to infer that `attachContext` is using a `HierarchyRequest` type request... however we're already doing a lot of fancy things here and I couldn't find a simple way to do it... 

Personally I don't mind it so much as this reads nicely to me as: Use this request, for this path, handled by this logic

Reply:
Agreed! We can always refactor it in the future if either out TS skills or TS itself gets enhanced!